From c13d921ec5d8486be3ba2b6dd37c6fbc72fad2da Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca.weiss@fairphone.com>
Date: Sun, 4 Jan 2026 15:22:31 +0100
Subject: [PATCH] Fix build with Qt5 5.15.10

This partially reverts commit be7157e14 ("Fixes in qmake2cmake based
cmake conversion").
---
 src/LomiriToolkit/ucstylehints.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/LomiriToolkit/ucstylehints.cpp b/src/LomiriToolkit/ucstylehints.cpp
index b8e337f72..fccbea308 100644
--- a/src/LomiriToolkit/ucstylehints.cpp
+++ b/src/LomiriToolkit/ucstylehints.cpp
@@ -104,8 +104,13 @@ void UCStyleHints::decodeBinding(const QString &propertyPrefix, const QQmlRefPoi
     QString propertyName = propertyPrefix + compilationUnit->stringAt(binding->propertyNameIndex);
 
     // handle grouped properties first
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 11)
     if (binding->type() == QV4::CompiledData::Binding::Type_GroupProperty
         || binding->type() == QV4::CompiledData::Binding::Type_AttachedProperty) {
+#else
+    if (binding->type == QV4::CompiledData::Binding::Type_GroupProperty
+        || binding->type == QV4::CompiledData::Binding::Type_AttachedProperty) {
+#endif
 
         const QV4::CompiledData::Object *subObj = compilationUnit->objectAt(binding->value.objectIndex);
         const QV4::CompiledData::Binding *subBinding = subObj->bindingTable();
@@ -116,7 +121,11 @@ void UCStyleHints::decodeBinding(const QString &propertyPrefix, const QQmlRefPoi
         return;
     }
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 11)
     switch (binding->type()) {
+#else
+    switch (binding->type) {
+#endif
     case QV4::CompiledData::Binding::Type_Script:
     {
         QString expression = compilationUnit->bindingValueAsScriptString(binding);
-- 
2.52.0

